### PR TITLE
RFR - Fix sudo in Paramiko remote script runner

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
Tried a bunch of approaches and settled on this approach. (See PR https://github.com/StackStorm/st2/pull/1824)

```
(virtualenv)/m/s/s/st2 git:paramiko_remote_script_sudo ❯❯❯ st2 run fixtures.streamwriter-script-remote hosts=localhost stream=stdout sudo=false
.
id: 55ce3efe32ed354d080341a1
status: succeeded
result:
{
    "localhost": {
        "succeeded": true,
        "failed": false,
        "return_code": 0,
        "stderr": "",
        "stdout": "STREAM IS STDOUT."
    }
}
(virtualenv)/m/s/s/st2 git:paramiko_remote_script_sudo ❯❯❯ st2 run fixtures.streamwriter-script-remote hosts=localhost stream=stdout sudo=true
.
id: 55ce3f0532ed354d080341a3
status: succeeded
result:
{
    "localhost": {
        "succeeded": true,
        "failed": false,
        "return_code": 0,
        "stderr": "",
        "stdout": "STREAM IS STDOUT."
    }
}
(virtualenv)/m/s/s/st2 git:paramiko_remote_script_sudo ❯❯❯
```

### CODE REVIEW

**Comment by Kami** [comment]
LGTM

File: st2common/st2common/models/system/paramiko_script_action.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+        if self.sudo:
+            command = 'sudo -E -- bash -c %s' % command
+
+        return command
+
+    @staticmethod
+    def _get_quoted_command_string(remote_script_path, script_arguments):
+        if script_arguments:
+            command = quote_unix('%s %s' % (remote_script_path, script_arguments))
```
Comment (Kami) [inline st2common/st2common/models/system/paramiko_script_action.py]: I assume each script argument is already quoted in `_get_script_arguments`, right?

Comment (lakshmi-kannan) [inline st2common/st2common/models/system/paramiko_script_action.py]: Yes, they are.

File: st2common/tests/unit/test_paramiko_script_action_model.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+class ParamikoRemoteScriptActionTests(unittest2.TestCase):
+
+    def test_get_command_string(self):
+        local_script_path = '/opt/stackstorm/packs/fixtures/actions/remote_script.sh'
+        script_action = ParamikoRemoteScriptAction('fixtures.remote_script',
+                                                   '55ce39d532ed3543aecbe71d',
+                                                   local_script_path,
+                                                   '/opt/stackstorm/packs/fixtures/actions/lib/',
+                                                   named_args={'stream': 'stdout'},
```
Comment (Kami) [inline st2common/tests/unit/test_paramiko_script_action_model.py]: Would be good to add whitespace to the argument value (or similar) and make sure the argument itself is also quoted correctly.

Besides that, LGTM.

**Comment by Kami** [comment]
Also, a better "example" in the description would be using `whoami` or similar :P

**Comment by lakshmi-kannan** [comment]
>Also, a better "example" in the description would be using whoami or similar :P

That's a remote command. Problem is with a script.

**Comment by Kami** [comment]
@lakshmi-kannan Script could run `whoami`, anyways :P

**Pull request merged**